mkdocs/docs/reference/experiment.md

```diff
@@ -304,3 +305,4 @@ Standard Kubernetes [meta.v1/ObjectMeta](https://kubernetes.io/docs/reference/ge
 | value | string | Value of a variable. | Yes |
 
 [^1]: `A/B/n` experiments involve more than one candidate. Their description is coming soon.
+[^1]: This feature is built using the [`expr` Go expression library](https://github.com/antonmedv/expr). The [Experiment Go struct](https://pkg.go.dev/github.com/iter8-tools/etc3@v0.1.30/api/v2alpha2#Experiment) is used as the environment for expression evaluation. Hence, `.Status.Analysis != nil` is also a valid condition.
```

Comment:
I think `.Status.Analysis` is preferable to `.this.Status.Analysis` since it is shorter. Perhaps we can work towards getting rid of `.this` in the variable interpolation as well?

It sort of makes sense that the default source of data for interpolation is the experiment struct.

Reply:
Besides not that `CandidateWon()`, `not CandidateWon()`, `WinnerFound()`, and `not WinnerFound()` are enough to cover almost all the use-cases we have currently documented within the tutorials.

I think `.Status.Analysis....` itself is less preferable to meaningful builtins such as the above.

Reply:
With the new task specific versionInfo, we have more options opening up ... we can extend tasks to do the following.
```yaml
- name: lib/task
  condition: <your condition here>
  for: WinnerOrBaseline() # Winner(),  Baseline(), Candidate()  not Winner() are some other options
  versionInfo:
  - variables:
  ...
```

This gets rid of interpolation altogether and simply selects the version(s) **for** which the task will run. It will use the selected version's versionInfo.

Reply:
Actually, we can do one better ... 

```yaml
- name: common/shell
  condition: <your condition here>
  with:
    script: kubectl apply -f {{ .WinnerOrBaseline() | .manifest }}
    versionInfo:
    - variables: # for baseline
      - name: manifest
        value: https://baseline.manifest.yaml
    - variables: # for candidate
      - name: manifest
        value: https://candidate.manifest.yaml
```

Reply:
It would be nice to get rid of version recommended for promotion altogether. Because, users need to understand **two** separate concepts, winner, as well as version recommended for promotion instead of only one. The above accomplishes this simplification.

Reply:
Issue created: https://github.com/iter8-tools/iter8/issues/945

Reply:
> I think `.Status.Analysis` is preferable to `.this.Status.Analysis` since it is shorter. Perhaps we can work towards getting rid of `.this` in the variable interpolation as well?
> 
> It sort of makes sense that the default source of data for interpolation is the experiment struct.

I am a bit ambivalent. I chose `.this` because I wanted to avoid possible conflicts with user variable names. While `.metadata` and `.spec` do seem unlikely, I thought `.status` seemed as likely as any other variable name.  `.this` seemed safe.

Agree that a couple of well defined functions will go a long way to making this irrelevant. I think we still need to learn which variables are most likely to be read.

lgtm